lib/grpc/GrpcServer.ts

```diff
@@ -36,9 +36,12 @@ class GrpcServer {
    */
   public listen = (port: number, host: string) => {
     assert(Number.isInteger(port) && port > 1023 && port < 65536, 'port must be an integer between 1024 and 65535');
-
     try {
-      this.server.bind(`${host}:${port}`, grpc.ServerCredentials.createInsecure());
+      const bindCode = this.server.bind(`${host}:${port}`, grpc.ServerCredentials.createInsecure());
+      if (bindCode !== port) {
+        this.logger.error(`Failed to listen on port: ${port}`);
+        throw new Error(`gRPC couldn't bind on port: ${port}`);
```

Comment:
Let's use pre-defined errors, defined per-domain, same as in here, so we can easily track them and give them codes in a later stage.